torchgeo/datasets/geo.py

```diff
@@ -329,7 +329,7 @@ def dtype(self) -> torch.dtype:
 
     def __init__(
         self,
-        root: str = "data",
+        paths: Union[str, list[str]] = "data",
```

Comment:
Can we use `Iterable` instead of `list`? I don't want to tell users they can't use a tuple if they want to. We may also want to support generators someday for online learning. I'm trying to get better at making these types as broad as possible.